aiohttp/web_app.py

```diff
@@ -430,7 +430,7 @@ async def _on_cleanup(self, app: Application) -> None:
                 await it.__anext__()
             except StopAsyncIteration:
                 pass
-            except Exception as exc:
+            except (Exception, asyncio.CancelledError) as exc:
```

Comment:
Is there a concern we would consume CancelledError if the current task was cancelling?

ie do we need?
```
if asyncio.current_task().cancelling() > 0:
   raise
```

Or is it ok to continue on ... maybe ok to consume it since its shutting down anyways and this isn't something we need to worry about?

Reply:
I feel like if we've stopped the cleanup context from a cancellation, we'd still want to run the other contexts. If we were to reraise, I'd suggest doing after the loop. Not sure if that's really worth doing though.

My bigger question is should we catch BaseException, so for example, KeyboardInterrupt can interrupt each cleanup context one-at-a-time?

Reply:
Usually, when I catch keyboard interrupted, I log something about trying to continue to shut down gracefully, and let them press it again to shut down now.

Reply:
Well, this is probably the second interrupt already. Question is whether it should require upto number of contexts + 1 to close, or stick with the current 2.

Reply:
I’d stick with the current two, as it’s confusing behavior to have to hit it many many times and the user might think something is broken, or hung